crates/miden-objects/src/note/assets.rs

```diff
@@ -107,6 +108,43 @@ impl NoteAssets {
         padded_assets
     }
 
+    /// Returns all Fungible Assets as an Iterator.
+    pub fn iter_fungible(&self) -> impl Iterator<Item = FungibleAsset> {
+        self.assets.iter().filter_map(|asset| {
+            if asset.is_fungible() {
+                Some(asset.unwrap_fungible())
+            } else {
+                None
+            }
+        })
+    }
+
+    /// Returns all NonFungible Assets as an Iterator.
+    pub fn iter_non_fungible(&self) -> impl Iterator<Item = NonFungibleAsset> {
+        self.assets.iter().filter_map(|asset| {
+            if asset.is_non_fungible() {
+                Some(asset.unwrap_non_fungible())
+            } else {
+                None
+            }
+        })
+    }
+
+    /// Returns total amount of fungible assets for a specific `Faucet ID`.
+    pub fn amount(&self, faucet_id: AccountId) -> u64 {
```

Comment:
I like this method, but I think the consensus from the issue seems to be more on not having it.

With `iter_fungible` it's also very easy to write this now, e.g.:

```rust
self.iter_fungible().fold(0, |acc, fungible_asset| {
    if fungible_asset.faucet_id() == faucet_id {
        acc + fungible_asset.amount()
    } else {
        acc
    }
})
```

So let's remove it from here.